test/controllers/image_controller_test.rb

```diff
@@ -485,6 +640,82 @@ def test_update_image
     assert_equal(new_name, image.reload.original_name)
   end
 
+  def test_update_image_no_changes
+    image = images(:agaricus_campestris_image)
+    params = {
+      "id" => image.id,
+      "image" => {
+        "when(1i)" => image.when.year.to_s,
+        "when(2i)" => image.when.month.to_s,
+        "when(3i)" => image.when.day.to_s,
+        "copyright_holder" => image.copyright_holder,
+        "notes" => image.notes,
+        "original_name" => image.original_name,
+        "license" => image.license
+      }
+    }
+
+    post_requires_login(:edit_image, params)
+
+    assert_flash_text(:runtime_no_changes.l,
+                      "Flash should say no changes " \
+                      "if no changes made when editing image")
+  end
+
+  # Prove that user can remove image from project
+  # by updating image without changes
+  def test_update_image_unchanged_remove_from_project
+    project = projects(:bolete_project)
+    assert(project.images.present?,
+           "Test needs Project fixture that has an Image")
+    image = project.images.first
+    user = image.user
+    params = {
+      "id" => image.id,
+      "image" => {
+        "when(1i)" => image.when.year.to_s,
+        "when(2i)" => image.when.month.to_s,
+        "when(3i)" => image.when.day.to_s,
+        "copyright_holder" => image.copyright_holder,
+        "notes" => image.notes,
+        "original_name" => image.original_name,
+        "license" => image.license
+      },
+      project: project
+    }
+    login(user.login)
+
+    post(:edit_image, params: params)
+
+    assert(project.reload.images.exclude?(image),
+           "Failed to remove image from project")
+  end
+
+  def test_update_image_save_fail
+    image = images(:turned_over_image)
+    assert_not_empty(image.projects,
+                     "Use Image fixture with a Project for best coverage")
+    params = {
+      "id" => image.id,
+      "image" => {
+        "when(1i)" => "2001",
+        "when(2i)" => "5",
+        "when(3i)" => "12",
+        "copyright_holder" => "Rolf Singer",
+        "notes" => "",
+        "original_name" => "new name"
+      }
+    }
+
+    login(image.user.login)
+    Image.any_instance.stubs(:save).returns(false)
```

Comment:
@pellaea: Is there an image (or pseudo-image) file which will make `@image.save` fail?
(I'd prefer to avoid stubbing Image#save if that can be done easily.)

Reply:
Sorry, no, I think the only way to get Image.save to fail is to stub something.  I suppose you could also stub validate_upload?  Or maybe stub upload_length to be huge or something like that?  You might also try uploading a plain old text file or something like that?